File: src/domain/entities/MonitoringEntities.ts
export type UserRole = 'student' | 'parent' | 'teacher' | 'admin';

export interface TeacherInvitationSummary {
  id: string;
  codePrefix: string;
  schoolName: string;
  maxUses: number;
  usedCount: number;
  expiresAt: string;
  status: 'active' | 'expired' | 'revoked' | 'exhausted';
  createdAt: string;
}

export interface ParentStudentLink {
  id: string;
  parentUserId: string;
  studentUserId?: string;
  linkCode: string;
  status: 'pending' | 'active' | 'revoked' | 'expired';
  expiresAt: string;
  createdAt: string;
}

export interface TeacherClass {
  id: string;
  teacherUserId: string;
  name: string;
  subject: string;
  gradeLevel: string;
  classCode: string;
  studentCount?: number;
  averageMastery?: number;
  createdAt: string;
}

export interface ClassMember {
  id: string;
  classId: string;
  studentUserId: string;
  status: 'active' | 'archived' | 'left';
  joinedAt: string;
}

export interface LearningSession {
  id: string;
  userId: string;
  courseId: string;
  lessonId?: string;
  startedAt: string;
  lastHeartbeatAt: string;
  endedAt?: string;
  activeSeconds: number;
  platform: 'web' | 'android' | 'ios';
}

export interface DailyLearningStats {
  id: string;
  userId: string;
  activityDate: string; // YYYY-MM-DD
  activeSeconds: number;
  lessonsCompleted: number;
  questionsAnswered: number;
  correctAnswers: number;
  accuracyPercent: number;
  xpEarned: number;
  overallScore: number;
}

export interface WeeklyActivityDay {
  dayName: string; // "Dushanba", "Seshanba", etc.
  dateStr: string;
  activeMinutes: number;
  lessonsCount: number;
  accuracyPercent: number;
}

export interface StudentAlert {
  id: string;
  userId: string;
  alertType: 'inactivity' | 'weak_topic' | 'goal_missed' | 'streak_risk';
  title: string;
  description: string;
  severity: 'info' | 'warning' | 'critical';
  metadata?: Record<string, unknown>;
  isResolved: boolean;
  createdAt: string;
}

export interface ChildSummary {
  studentId: string;
  firstName: string;
  lastName: string;
  displayName: string;
  avatarUrl?: string;
  todayActiveMinutes: number;
  overallScore: number;
  streakDays: number;
  xp: number;
  weakestSkillName: string;
  weakestSkillScore: number;
  strongestSkillName: string;
  strongestSkillScore: number;
  todayGoalMinutes: number;
  goalCompletionPercent: number;
  statusTitle: string;
  pedagogicalAdvice: string;
}

export interface ClassStudentSummary {
  studentId: string;
  name: string;
  email: string;
  avatarUrl?: string;
  todayActiveMinutes: number;
  overallScore: number;
  completedLessonsCount: number;
  totalAttemptsCount: number;
  weakestSkillName: string;
  weakestSkillScore: number;
  status: 'Yaxshi' | 'Nazorat' | 'E’tibor' | 'A’lo';
  alertCount: number;
  lastActiveDate: string;
}

File: mobile/lib/domain/entities/monitoring_entities.dart
enum UserRole { student, parent, teacher, admin }

class ParentStudentLink {
  final String id;
  final String parentUserId;
  final String? studentUserId;
  final String linkCode;
  final String status;
  final DateTime expiresAt;
  final DateTime createdAt;

  const ParentStudentLink({
    required this.id,
    required this.parentUserId,
    this.studentUserId,
    required this.linkCode,
    required this.status,
    required this.expiresAt,
    required this.createdAt,
  });
}

class TeacherClass {
  final String id;
  final String teacherUserId;
  final String name;
  final String subject;
  final String gradeLevel;
  final String classCode;
  final int studentCount;
  final int averageMastery;
  final DateTime createdAt;

  const TeacherClass({
    required this.id,
    required this.teacherUserId,
    required this.name,
    required this.subject,
    required this.gradeLevel,
    required this.classCode,
    this.studentCount = 0,
    this.averageMastery = 0,
    required this.createdAt,
  });
}

class ChildSummary {
  final String studentId;
  final String firstName;
  final String lastName;
  final String displayName;
  final String avatarUrl;
  final int todayActiveMinutes;
  final int overallScore;
  final int streakDays;
  final int xp;
  final String weakestSkillName;
  final int weakestSkillScore;
  final String strongestSkillName;
  final int strongestSkillScore;
  final int todayGoalMinutes;
  final int goalCompletionPercent;
  final String statusTitle;
  final String pedagogicalAdvice;

  const ChildSummary({
    required this.studentId,
    required this.firstName,
    required this.lastName,
    required this.displayName,
    this.avatarUrl = '',
    required this.todayActiveMinutes,
    required this.overallScore,
    required this.streakDays,
    required this.xp,
    required this.weakestSkillName,
    required this.weakestSkillScore,
    required this.strongestSkillName,
    required this.strongestSkillScore,
    required this.todayGoalMinutes,
    required this.goalCompletionPercent,
    required this.statusTitle,
    required this.pedagogicalAdvice,
  });
}

class ClassStudentSummary {
  final String studentId;
  final String name;
  final String email;
  final int todayActiveMinutes;
  final int overallScore;
  final int completedLessonsCount;
  final int totalAttemptsCount;
  final String weakestSkillName;
  final int weakestSkillScore;
  final String status;
  final int alertCount;
  final String lastActiveDate;

  const ClassStudentSummary({
    required this.studentId,
    required this.name,
    this.email = '',
    required this.todayActiveMinutes,
    required this.overallScore,
    required this.completedLessonsCount,
    required this.totalAttemptsCount,
    required this.weakestSkillName,
    required this.weakestSkillScore,
    required this.status,
    required this.alertCount,
    required this.lastActiveDate,
  });
}

class WeeklyActivityDay {
  final String dayName;
  final String dateStr;
  final int activeMinutes;
  final int lessonsCount;
  final int accuracyPercent;

  const WeeklyActivityDay({
    required this.dayName,
    required this.dateStr,
    required this.activeMinutes,
    required this.lessonsCount,
    required this.accuracyPercent,
  });
}
